### PR TITLE
docs: Update README.md with more information about table creation, permissions, TTL, and logging.

### PR DESCRIPTION
**Pull request by ashovlin**
*Issue #, if available:* DOTNET-7002

*Description of changes:* Updates the README with more information about table creation, permissions, TTL, and logging.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +60,76 @@ var app = builder.Build();
...
+    }
+```
+# Time to Live
+`IDistributedCache` exposes several options for for expiring entries from the cache on its [`DistributedCacheEntryOptions`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.distributed.distributedcacheentryoptions). If using the `IDistributedCache` to store session state for an ASP.NET Core application, [`SessionOptions.IdleTimeout`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.sessionoptions.idletimeout) controls the expiration for cache entries.
+
+The AWS .NET Distributed Cache Provider relies on DynamoDB's [Time to Live (TTL) feature](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/TTL.html) to delete expired items from the table. TTL will be enabled automatically when tables are created via `CreateTableIfNotExists`. If you provide an existing table and TTL is **not enabled, expired entries will not be deleted** and a warning will be logged during the first cache operation.
+
+Because DynamoDB's TTL feature can take up to a few days to delete expired items, the cache provider filters out items that have expired but not been deleted yet on the client side.
+
+# Monitoring
```
Comment (normj) [inline README.md:128]: I think I would replace `Monitoring` with `Auditing`. When I think of `Monitoring` I think of CloudWatch not CloudTrail.

**Review by normj — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -62,6 +60,76 @@ var app = builder.Build();
...
+* The [on-demand](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadWriteCapacityMode.html#HowItWorks.OnDemand)  capacity mode
+* [Time to Live](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/TTL.html) enabled
+* Encryption using an [AWS KMS key owned and managed by DynamoDB](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/encryption.howitworks.html#ddb-owned). If you would like to use an AWS managed or customer managed key, you can update the table after it is created or provide your own table. See [Managing encrypted tables](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/encryption.tutorial.html) in the Amazon DynamoDB Developer Guide.
+
+ It is recommended to only set `CreateTableIfNotExists` to `true` in development environments. Production environments should create the table before deployment. This will allow production applications to require fewer permissions and have a faster startup.
+
+# Permissions
```
[the remark this answers is not in the document]
Comment (ashovlin) [reply README.md:73]: Agreed, rewrote this paragraph. I also moved it up above the description of what happens when the table is created.

File: README.md [hunk 3]
```diff
@@ -62,6 +60,76 @@ var app = builder.Build();
...
+To use the cache provider with an _existing_ DynamoDB table, the following permissions are required:
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:PutItem",
```
Comment (normj) [inline README.md:82]: Following the comment you added about the recommended is `CreateTableIfNotExists` to `false` and providing both a `PartitionKeyName` and `TTLAttributeName`. If they do that then the library does not require `dynamodb:DescribeTable` and `dynamodb:DescribeTimeToLive`. Can we remove those 2 permissions from this policy.

**Review by normj — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (96malhar) [reply README.md:82]: You can add a sentence at the bottom of the snippet saying - " If the `PartitionKeyName` and `TTLAttributeName` are set, then the `dynamodb:DescribeTable` and `dynamodb:DescribeTimeToLive` permissions can be omitted.

Comment (ashovlin) [reply README.md:82]: Addressed via https://github.com/awslabs/aws-dotnet-distributed-cache-provider/compare/2a4b15819408cd014c9bff1025fc0b3bccf13c0c..94d6ed5e7e59d3bbb28cfa1644dff55dfd15f4e0.

I added a third block for the middle case (existing table, but unknown key names)

**Review by normj — APPROVED** [review]

**Review by 96malhar — APPROVED** [review]

**Pull request merged**